src/main/java/io/cdap/e2e/pages/actions/CdfPluginPropertiesActions.java

```diff
@@ -290,6 +291,36 @@ public static Map<String, String> getOutputSchema() {
     return actualOutputSchema;
   }
 
+  /**
+   * Scroll down until Field-Type entry is found in the Output Schema table
+   *
+   * @param fieldName
+   * @param fieldType
+   * @return
+   */
+  public static Boolean scrollDownInOutputSchemaUntilFieldEntry(String fieldName, String fieldType) {
+    By locator = CdfPluginPropertiesLocators.locatorOfOutputSchemaFieldTypeEntry(fieldName, fieldType);
+    boolean isFieldTypeEntryPresent = WaitHelper.waitForElementToBeOptionallyPresent(locator, 1);
+    int totalSchemaRowsInit = CdfPluginPropertiesLocators.outputSchemaRows.size();
+    int totalSchemaRowsUpdated;
+
+    while (!isFieldTypeEntryPresent) {
```

Comment:
why do we need this extra for loop?
When is it possible that we need to traverse the whole output schema?

Reply:
We came across a few use cases where the list of Output Schema fields is longer than 50. 
The current application behaviour is such that it only shows the first ~20 fields and only shows the rest of the fields once we start scrolling down. The fields at the bottom are not even present in the DOM. 

Hence, we are checking for the presence of the field and if it is not available then only scrolling down and checking for its presence again.

Reply:
I see, thanks for the info. Could you please add a comment in the code block?